### DISCUSSION TITLE
Perform Auto-Type button needs a new variant

### DISCUSSION THREAD

**Discussion by KodeXall** [keepassxreboot/keepassxc#9260] [Ideas]
I propose adding [not as merely as new variant but just] a new button, next to current `Perform Auto-Type` - the `Perform Auto-Opening And Auto-Type` shortcut

As it name is self-explaining, using it would result with saving users time and clicks, because they no longer would have to first click a link in database and then go back to it only just to click `Perform Auto-Type`. This would reduce the annoyance coming out of tiresome and repetitive tasks

But maybe this new option / buton could also have some kind of detection mechanism that would wait for the page to really load, before executing its later part? I myself have issues with my Internet connection, which sometimes just slows down why other times slows down and then cuts me off entirely from the Internet

**Comment by droidmonkey** [comment]
This is why we have the browser extension and global Auto-Type shortcut..

**Reply by KodeXall** [reply]
Either I just do not get it or something / sometimes is not working properly on my setup

Because for example on this site https://www.voidtools.com/forum/ucp.php?mode=login all I get is a grayed out _**KeePassXC**_ icon in the `Username` field - while the add-on has button `Redirect login fields`, afer pressing of which however nothing happens. And if I execute my `Global Auto-Type` keyboard shortcut then in the _**Username**_ field on this page what is written is the password. And yes: credentials for it in my database are correct, because clicking `Perform Auto-Type` always has been working A-OK for this site. And no: other forums that run on the _**phpBB Forum Software**_ and which I login into using _**KeePassXC**_ do not give me such issues. It is only the forum of _**Everything**_ that is forcing me to go twice to my database, one for opening of the site and second for getting logged in to it. And this has been a long time issue of mine, despite numeral upgrades of _**Firefox**_, _**KeePassXC-Browser**_ add-on and _**KeePassXC**_ itself

Other annoying example is _**ChatGPT**_. The page that shows up afer clicking `Log in` on https://chat.openai.com/auth/login also has grayed out icon of _**KeePassXC**_. Although this I could understand, as the address before and after clicking of that `Log in` button changes, right? But when I press my `Global Auto-Type` shortcut then what I get is his pop-up window from _**KeePassXC**_ containing list of all of my entries in my database. And as they are many I have to either scroll down or narrow down the list using the `Search` box at its bottom. which is just inconvenient

Thus a need for another alternative way, one more option available for users - the proposed by me `Perform Auto-Opening And Auto-Type` button

**Reply by droidmonkey** [reply]
Works fine for me with a new entry:

![image](https://user-images.githubusercontent.com/2809491/226203417-26e580a6-d6b8-4142-9f21-419ec1333e0d.png)

Make sure you URL is correct on the entry (https://voidtools.com)

For OpenAI you need to do this:

![image](https://user-images.githubusercontent.com/2809491/226203480-6e2f3da3-c828-4264-9ed1-f75aca5ed58e.png)

**Comment by KodeXall** [comment]
I have deleted my _**Everything**_ entry from my database and closed it along with **_Firefox_**. Then I reopened _**KeePassXC**_ and created entry for it from scratch, pasting in all previous data from a temporarily TXT file - except for _**Title**_, because I gave it a brand new one, so that o reduce a possibility of potential overlapping

But despite this I still have a grayed out _**KPXC**_ icon and do not see this blue button for adding poitned out to me with red arrow. I was only able to manually choose for the add-on what box is suppose o be the `Username` and which one the `Password` - and sill it did not change a thing in how my setup works

**Comment by KodeXall** [comment]
Now I can use the button from add-on on forum of Everything. Did installing a newer version of Firefox somehow unblocked this? I do not know

As for me not seeing the KeePassXC icon in the login field at all: it was due to me using a makeshift CSS adjustment on this forum. After pin pointing the culprit in my code I now get to see and use it. So this one was a false alarm, which I should have thought about before even reporting his issue- sorry for that

I also took care of the ChatGPT issue and learnt something from it

And so, at least for now, case closed

Thank you for yor time
